File: WondayWall.iOS/WondayWall/Models/AppConfig.swift
import Foundation

// アプリ設定モデル
struct AppConfig: Codable {
    // 取得対象のカレンダー識別子一覧（EKCalendar.calendarIdentifier）
    var targetCalendarIds: [String] = []
    // RSS ソース URL 一覧
    var rssSources: [String] = []
    // ユーザー追加プロンプト
    var userPrompt: String = ""
    // 自動更新スケジュール
    var schedule: UpdateSchedule = .onceADay
    // 自動生成有効/無効
    var autoGenerationEnabled: Bool = true
    // 変化がなければ生成をスキップする
    var skipIfNoChanges: Bool = false
    // 通知を有効にする
    var notificationsEnabled: Bool = true
    // Wi-Fi 接続時のみ生成する
    var wifiOnlyGeneration: Bool = false
    // 手動生成でも Flex ティアを強制使用する
    var forceFlexTier: Bool = false
    // WondayWall アルバムの最大保存枚数（超過分は古い順にアルバムから外す）
    var albumMaxCount: Int = 1
    // 初回セットアップ完了済みかどうか
    var hasCompletedInitialSetup: Bool = false
    // 初回セットアップ時に保存した画面のネイティブピクセルサイズ
    var screenNativePixelWidth: Double?
    var screenNativePixelHeight: Double?
}

File: WondayWall.Android/app/src/main/kotlin/com/studiofreesia/wondaywall/models/AppConfig.kt
package com.studiofreesia.wondaywall.models

import kotlinx.serialization.Serializable

// アプリ設定モデル（Windows版 AppConfig と同名フィールドで揃える）
@Serializable
data class AppConfig(
    // Google AI APIキー
    val googleAiApiKey: String = "",
    // 取得対象カレンダーID一覧（Calendar ProviderのカレンダーID）
    val targetCalendarIds: List<String> = emptyList(),
    // RSSソースURL一覧
    val rssSources: List<String> = emptyList(),
    // ユーザーの追加プロンプト・制約
    val userPrompt: String = "",
    // 自動更新スケジュール（Windows版 AppConfig.Schedule と同名）
    val schedule: UpdateSchedule = UpdateSchedule.OnceADay,
    // 生成材料に変化がなければスキップするか
    val skipGenerationWhenNoChanges: Boolean = false,
    // ロック画面も更新するか（false = ホーム画面のみ）
    val updateLockScreen: Boolean = false,

    // Android固有設定
    // 自動生成を有効にするか
    val autoGenerationEnabled: Boolean = false,
    // 生成時に通知するか（成功・失敗ともに同一フラグで管理）
    val showNotification: Boolean = true,
    // Wi-Fi接続時のみ生成するか
    val generateOnlyOnWifi: Boolean = false,
    // 省電力モード時は生成をスキップするか
    val skipOnBatterySaver: Boolean = true,
    // 手動生成でも Flex ティアを強制使用する
    val forceFlexTier: Boolean = false,
)
